provisioner/puppet-masterless/provisioner.go

```diff
@@ -296,8 +299,16 @@ func (p *Provisioner) Provision(ui packer.Ui, comm packer.Communicator) error {
 		PuppetBinDir:    p.config.PuppetBinDir,
 		Sudo:            !p.config.PreventSudo,
 		WorkingDir:      p.config.WorkingDir,
-		ExtraArguments:  strings.Join(p.config.ExtraArguments, " "),
+		ExtraArguments:  "",
+	}
+
+	p.config.ctx.Data = &data
+	_ExtraArguments, err := interpolate.Render(strings.Join(p.config.ExtraArguments, " "), &p.config.ctx)
```

Comment:
So what benefit do you gain from interpolating the ExtraArguments separately instead of at the same time? Casually reading this section of code it looks like the outcome will be the same.